<?xml version="1.0"?>
<launch>
    <!-- Launch RVIZ  -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find deformable_manipulations_fabric)/rviz/fabric_sim_test.rviz" output="screen"/>

    <!-- Launch the fabric Simulation that reads the location data of robots and attaches the fabric corners to those locations  -->
    <!-- REPRESENTS THE REAL fabric-->
    <node type="fabric_simulator_node" name="fabric_simulator_node" pkg="fabric_simulator" output="screen">
        <rosparam command="load" file="$(find deformable_manipulations_fabric)/config/fabric_simulator.yaml" subst_value="true" />
    </node>

    <!-- Launch the Distance Publisher between the fabric and the specified Polhedron -->
    <node type="fabric_to_polyhedron_distance.py" name="fabric_to_polyhedron_distance" pkg="deformable_manipulations_fabric" output="screen">
        <rosparam command="load" file="$(find deformable_manipulations_fabric)/config/fabric_to_polyhedron_distance.yaml" subst_value="true" />
    </node>

    <!-- SPACE MOUSE -->
    <include file="$(find spacenav_node)/launch/classic.launch"/>

    <node type="fake_odom_publisher_gui.py" name="fake_odom_publisher_gui" pkg="deformable_manipulations_fabric" output="screen">
        <!-- <rosparam command="load" file="$(find deformable_manipulations_fabric)/config/fabric_to_polyhedron_distance.yaml" subst_value="true" /> -->
    </node>

    <node type="fabric_position_controller.py" name="fabric_position_controller_node" pkg="deformable_manipulations_fabric" output="screen">
        <!-- <rosparam command="load" file="$(find deformable_manipulations_fabric)/config/dlo_to_polyhedron_distance.yaml" subst_value="true" /> -->
    </node>
    
</launch>